tree/treeplayer/src/TSimpleAnalysis.cxx

```diff
@@ -0,0 +1,384 @@
+#histograms. They have to be the follow syntax:
+#name = expression if cut
+#which corresponds to chain->Draw("expression >> name", "cut")
+#i.e. it will create a histogram called name and store it in
+#file_output.root.
+#"if cut" is optional
+hpx=px if px<-3   #first histogram
+hpxpy=px:py    #second histogram
+
+#End of the configuration file
+```
+*/
+
+////////////////////////////////////////////////////////////////////////////////
+/// Delete leading and trailing white spaces in a string.
+///
+/// param[in] line line read from the input file
+
+static void DeleteSpaces(std::string& line)
+{
+   // Delete leading spaces
+   std::size_t firstNotSpace = line.find_first_not_of(" \t");
+   if (firstNotSpace != std::string::npos)
+      line = line.substr(firstNotSpace, line.size() - firstNotSpace);
+   else
+      line.clear();
+   // Delete trailing spaces
+   std::size_t lastNotSpace = line.find_last_not_of(" \t");
+   if (lastNotSpace != std::string::npos)
+      line = line.substr(0, lastNotSpace + 1);
+}
+
+////////////////////////////////////////////////////////////////////////////////
+/// Handle the expression lines of the input file in order to pass the
+/// elements to the members of the object.
+///
+/// param[in] line line or read from the input file either the expression passed to the constructor
+
+std::string TSimpleAnalysis::HandleExpressionConfig(const std::string& line)
+{
+   static const std::string kCutIntr = " if ";
+
+   std::size_t equal = line.find("=");
+   if (equal == std::string::npos)
+      return "Error: missing '='";
+
+   // Set the histName value
+   std::string histName = line.substr(0, equal);
+   DeleteSpaces(histName);
+   if (histName.empty())
+      return "Error: no histName found";
+
+   //Set the histExpression value
+   std::size_t cutPos = line.find(kCutIntr, equal);
+   std::string histExpression = line.substr(equal + 1, cutPos - equal - 1);
+   DeleteSpaces(histExpression);
+   if (histExpression.empty())
+      return "Error: no expression found";
+
+   // Set the histCut value
+   std::string histCut;
+   if (cutPos != std::string::npos) {
+      histCut = line.substr(cutPos + kCutIntr.size());
+      DeleteSpaces(histCut);
+      if (histCut.empty())
+         return "Error: invalid syntax";
+   }
+   else
+      histCut = "";
+
+   // Set the map that contains the histName, histExpressions and histCut values
+   auto check = fHists.insert(std::make_pair((const std::string&)histName,
+                                             std::make_pair(histExpression, histCut)));
+
+   // Check if there are histograms with the same name
+   if (!check.second)
+      return "Duplicate histogram name";
+   return "";
+}
+
+////////////////////////////////////////////////////////////////////////////////
+/// Constructor for the case of command line parsing arguments. It sets the members
+/// of the object.
+///
+/// \param[in] output name of the output file
```

Comment:
Add a '-' after variable name, i.e. `\param[in] output - name of the output file`